dependencies/clients/teku.go

```diff
@@ -0,0 +1,329 @@
+	}
+
+	fullPath, err = utils.PrepareTimestampedFile(logFolder, t.CommandName())
+	if err != nil {
+		return
+	}
+
+	err = os.WriteFile(fullPath, []byte{}, 0750)
+	if err != nil {
+		return
+	}
+
+	logFile, err = os.OpenFile(fullPath, os.O_RDWR, 0750)
+	if err != nil {
+		return
+	}
+
+	command.Stdout = logFile
+	command.Stderr = logFile
+
+	log.Infof("🔄  Starting %s", t.Name())
+	err = command.Start()
+	if err != nil {
+		return
+	}
+
+	pidLocation := fmt.Sprintf("%s/%s.pid", pid.FileDir, t.CommandName())
+	err = pid.Create(pidLocation, command.Process.Pid)
+
+	time.Sleep(1 * time.Second)
+
+	log.Infof("✅  %s started!", t.Name())
+
+	return
+}
+
+func untarDir(dst string, t *tar.Reader) error {
+	for {
+		header, err := t.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return err
+		}
+
+		path := filepath.Join(dst, header.Name)
+
+		// for the sake of compatibility with updated versions remove the tag from the tarred file - teku/teku-xx.x.x => teku/teku, same with jdk
+		switch {
+		case strings.Contains(header.Name, tekuFolder):
+			newHeader := replaceRootFolderName(header.Name, tekuFolder)
+			path = filepath.Join(dst, newHeader)
+
+		case strings.Contains(header.Name, jdkFolder):
+			newHeader := replaceRootFolderName(header.Name, jdkFolder)
+			path = filepath.Join(dst, newHeader)
+		}
+
+		info := header.FileInfo()
+		if info.IsDir() {
+			err = os.MkdirAll(path, info.Mode())
+			if err != nil {
+				return err
+			}
+
+			continue
+		}
+
+		dir := filepath.Dir(path)
+		err = os.MkdirAll(dir, info.Mode())
+		if err != nil {
+			return err
+		}
+
+		file, err := os.OpenFile(path, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, info.Mode())
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+		_, err = io.Copy(file, t)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func setupJava(jdkURL string) (err error) {
+	log.Info("⬇️  Downloading JDK...")
+
+	err = installAndUntarFromURL(jdkURL)
+	if err != nil {
+		return err
+	}
+
+	log.Info("✅  JDK downloaded!\n\n")
+
+	luksoNodeDir, err := os.Getwd()
+	if err != nil {
+		return
+	}
+
+	javaHomeVal := fmt.Sprintf("%s/%s/%s", luksoNodeDir, tekuDepsFolder, jdkFolder)
+
+	log.Infof("⚙️  To continue working with Teku please export the JAVA_HOME environmental variable.\n"+
+		"The recommended way is to add the following line:\n\n"+
+		"export JAVA_HOME=%s\n\n"+
+		"To the bash startup file of your choosing (like .bashrc)", javaHomeVal)
+
+	return
+}
+
+func installAndUntarFromURL(url string) (err error) {
+	response, err := http.Get(url)
+	if nil != err {
+		return
+	}
+
+	defer func() {
+		_ = response.Body.Close()
+	}()
+
+	if response.StatusCode == http.StatusNotFound {
+		log.Warnf("⚠️  File under URL %s not found - skipping...", url)
+
+		return nil
+	}
+
+	if http.StatusOK != response.StatusCode {
+		return fmt.Errorf(
+			"❌  Invalid response when downloading on file URL: %s. Response code: %s",
```

Comment:
Invalid response while downloading the file at URL:

Reply:
Done